master/internal/rm/kubernetesrm/jobs.go

```diff
@@ -253,18 +253,22 @@ func newJobsService(
 }
 
 func (j *jobsService) syncNamespaces(ns []string) error {
-	err := j.startEventListeners(ns)
-	if err != nil {
-		return err
-	}
-
-	err = j.startPreemptionListeners(ns)
-	if err != nil {
-		return err
-	}
-
 	for _, namespace := range ns {
+		if _, ok := j.namespacesWithInformers[namespace]; ok {
+			continue
+		}
 		j.namespacesWithInformers[namespace] = true
```

Comment:
If the `startEventListeners` below fails, the namespace wouldn't have an informer yet though, right? Should we only set `namespacesWithInformers` to true if no error is returned?

Reply:
Ooh this is a great point! Yepp I'll make this change